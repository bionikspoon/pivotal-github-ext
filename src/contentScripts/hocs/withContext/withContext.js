import React from 'react'
import { setDisplayName, wrapDisplayName } from 'recompose'

const withContext = (Consumer, propName) => BaseComponent => {
  const factory = React.createFactory(BaseComponent)

  function WithContext(props) {
    return (
      <Consumer>
        {context => factory({ ...props, [propName]: context })}
      </Consumer>
    )
  }

  if (process.env.NODE_ENV !== 'production') {
    return setDisplayName(wrapDisplayName(BaseComponent, 'withContext'))(
      WithContext
    )
  }
  return WithContext
}

export default withContext
