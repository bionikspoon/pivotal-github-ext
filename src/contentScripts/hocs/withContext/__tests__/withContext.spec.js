import { mount } from 'enzyme'
import React from 'react'
import withContext from '..'

describe('hocs/withContext', () => {
  const store = {
    getState: () => ({
      todos: ['eat', 'drink', 'sleep'],
      counter: 12,
    }),
  }
  const StoreContext = React.createContext(store)
  const Div = props => <div {...props} />

  let TestComponent = null

  beforeEach(() => {
    TestComponent = withContext(StoreContext.Consumer, 'store')(Div)
  })

  test('it sets display name', () => {
    expect(TestComponent.displayName).toEqual('withContext(Div)')
  })

  test('it matches snapshot', () => {
    const wrapper = mount(<TestComponent />)

    expect(wrapper).toMatchSnapshot()
  })

  test('it adds context to props', () => {
    const wrapper = mount(<TestComponent />)

    expect(wrapper.find(Div).prop('store')).toEqual(store)
  })
})
