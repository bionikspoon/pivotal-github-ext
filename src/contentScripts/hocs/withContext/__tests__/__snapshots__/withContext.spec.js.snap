// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hocs/withContext it matches snapshot 1`] = `
<withContext(Div)>
  <Div
    store={
      Object {
        "getState": [Function],
      }
    }
  >
    <div
      store={
        Object {
          "getState": [Function],
        }
      }
    />
  </Div>
</withContext(Div)>
`;
